lib/gringotts/gateways/securion_pay.ex

```diff
@@ -0,0 +1,217 @@
+
+  For reference see [SecurionPay's API (v1) documentation][docs].
+
+  The following set of functions for SecurionPay have been implemented:
+
+  | Action                                       | Method        |
+  | ------                                       | ------        |
+  | Authorize a Credit Card                      | `authorize/3` |
+
+  [home]: https://securionpay.com/
+  [docs]: https://securionpay.com/docs
+
+  ## The `opts` argument
+
+  Most `Gringotts` API calls accept an optional `keyword` list `opts` to supply
+  [optional arguments][extra-arg-docs] for transactions with the SecurionPay
+  gateway. The following keys are supported:
+
+  | Key                      | Remark                                                                                        |
+  | ----                     | ---                                                                                           |
+  | `customerId`             | Unique identifier of the customer                                                             |
+
+  ## Registering your SecurionPay account at `Gringotts`
+
+  After [making an account on SecurionPay][SP], find your `Secret key` at [Account Settings][api-key] 
+
+  Your Application config **must include the `:secret_key`  field**.
+  It would look something like this:
+
+      config :gringotts, Gringotts.Gateways.SecurionPay,
+        secret_key: "your_secret_key"
+
+  [SP]: https://securionpay.com/
+  [api-key]: https://securionpay.com/account-settings#api-keys
+
+  ## Supported countries
+
+  SecurionPay supports the countries listed [here][country-list]
+
+  [country-list]: https://securionpay.com/supported-countries-businesses/
+  """
+
+  @base_url "https://api.securionpay.com/"
+  # The Base module has the (abstract) public API, and some utility
+  # implementations.
+
+  use Gringotts.Gateways.Base
+
+  # The Adapter module provides the `validate_config/1`
+  # Add the keys that must be present in the Application config in the
+  # `required_config` list
+  use Gringotts.Adapter, required_config: [:secret_key]
+
+  import Poison, only: [decode: 1]
+
+  alias Gringotts.{CreditCard, Response, Address}
+
+  @doc """
+  Authorizes a credit card transaction.
+
+  The authorization validates the `card` details with the banking network,
+  places a hold on the transaction `amount` in the customer’s issuing bank and
+  also triggers risk management. Funds are not transferred.
+
+  The second argument can be a CreditCard or a cardId. The customerId of the cutomer who owns the card must be
+  given in optional field. 
+
+  To transfer the funds to merchant's account follow this up with a `capture/3`.
+
+  SecurionPay returns a `chargeId` which uniquely identifies a transaction (available in the `Response.id` field) 
+  which should be stored at your side to use in:
+
+  * `capture/3` an authorized transaction.
+  * `void/2` a transaction.
+
+  ## Example 1
+      iex> amount = Money.new(20, :USD}
```

Comment:
1. Typo `}`, and in line 81 there's a stray `:`
2. You can rewrite this section as:
```
## Examples
### With a `card_token` and `customer_token`
...
### With a `CreditCard` struct
...
```